.github/workflows/release.yml

```diff
@@ -34,7 +34,24 @@ jobs:
         run: sudo apt-get update && sudo apt-get install -y pandoc
 
       - name: Create release
-        run: ~/auto shipit
+        run: |
+          new_version="$(~/auto shipit --dry-run --quiet)"
+          if [ -z "$new_version" ]
+          then echo "Nothing to release"
+               exit
+          fi
+          changelog_version="$(perl -nle '/^# ([0-9]+\.\S*) \(.*\)/ and print $1 and exit' CHANGELOG.md)"
+          opts=
+          if dpkg --compare-versions "$new_version" lt "$changelog_version"
+          then echo "[ERROR] New version is less than latest version in changelog"
+               exit 1
+          elif dpkg --compare-versions "$new_version" eq "$changelog_version"
+          then opts=--no-changelog
+          elif ! dpkg --compare-versions "$new_version" gt "$changelog_version"
+          then echo "[ERROR] Could not compare new version '$new_version' to changelog version '$changelog_version'"
+               exit 1
+          fi
+          ~/auto shipit $opts
```

Comment:
elaborate but looks kosher... I was fearing that the empty `$opts` would freak out `auto` but my shell foo was weak -- should be ok unless it is `"$opts"`:
```
lena:/tmp
$> opts=                   

$> ./printargs.sh 1 2 $opts
> 1
> 2

$> ./printargs.sh 1 2 "$opts"
> 1
> 2
> 

$> cat printargs.sh 
#!/bin/bash
for a in "$@"; do
    echo "> $a"
done
```